### PR TITLE
libffi: hide platform-ignorant FFI ABI enum values

### PR DESCRIPTION
**Pull request by drcicero**
This change ignores processing the whole `ffitarget.h` file, which If i understood correctly contains platform-specific values, but only for one platform, and therefore is on most platforms wrong. I believe the platform-specific values can already be accessed `native` methods handwritten in the preset. I added the otherwise missing `FFI_TRAMPOLINE_SIZE()` method.

I think only two other variables remain in the `ffitarget.h` file, which are `FFI_CLOSURES = 1; FFI_GO_CLOSURES = 1;`. I'm not super sure, but i believe those are only there to enable/disable definitions in the other header, it shouldn't hurt not to have them.

I tried `mvn clean test --projects .,libffi` and it succeeds. I have not figured out if there are more tests to run?

See https://github.com/bytedeco/javacpp-presets/issues/1315

### CODE REVIEW

**Comment by drcicero** [comment]
Oh, the example https://github.com/bytedeco/javacpp/wiki/Mapping-Recipes#skipping-lines-from-header-files sounded like one can only skip lines of the form
```
// START COMPLEX DECLARATIONS
// ...
// END COMPLEX DECLARATIONS
```
but by studying the other InfoMap's I found it is actually possible to individual skip definition as well.

I suppose a simpler change would be to add the following:
```
.put(new Info("FFI_FIRST_ABI","FFI_SYSV","FFI_THISCALL","FFI_FASTCALL","FFI_STDCALL","FFI_PASCAL","FFI_REGISTER","FFI_MS_CDECL","FFI_LAST_ABI","FFI_DEFAULT_ABI").skip())
```

**Comment by drcicero** [comment]
Now, the change is only the small addition of a few skips (`.put(new Info("FFI_FIRST_ABI", "FFI_SYSV", "FFI_THISCALL", "FFI_FASTCALL", "FFI_STDCALL", "FFI_PASCAL", "FFI_REGISTER", "FFI_MS_CDECL", "FFI_LAST_ABI", "FFI_DEFAULT_ABI").skip())`), because those ones already are defined in the preset. And as this leads to having the correct `FFI_DEFAULT_ABI` definition autogenerated, I removed the manual definition of `FFI_DEFAULT_ABI` from the preset.

File: libffi/src/main/java/org/bytedeco/libffi/presets/ffi.java [hunk 1]
```diff
@@ -79,6 +79,9 @@ public void map(InfoMap infoMap) {
                .put(new Info("_ffi_type").pointerTypes("ffi_type"))
                .put(new Info("void (*)(void)").cast().pointerTypes("Pointer"))
                .put(new Info("__attribute__((deprecated))").annotations("@Deprecated"))
+               .put(new Info("FFI_FIRST_ABI", "FFI_SYSV", "FFI_THISCALL", "FFI_FASTCALL",
+                             "FFI_STDCALL", "FFI_PASCAL", "FFI_REGISTER", "FFI_MS_CDECL",
+                             "FFI_LAST_ABI", "FFI_DEFAULT_ABI").skip())
```
Comment (drcicero) [inline libffi/src/main/java/org/bytedeco/libffi/presets/ffi.java:84]: Actually, I did tell it to skip `FFI_DEFAULT_ABI`. But that didn't work, and I currently don't see why...

Comment (saudet) [reply libffi/src/main/java/org/bytedeco/libffi/presets/ffi.java:84]: Ah, ok, probably just a bug in JavaCPP, but it doesn't cause any problems, so whatever. :)

**Pull request merged**